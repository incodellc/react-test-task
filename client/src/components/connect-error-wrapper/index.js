import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;