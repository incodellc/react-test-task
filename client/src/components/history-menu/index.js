import HistoryMenu from './HistoryMenu';
export default HistoryMenu;