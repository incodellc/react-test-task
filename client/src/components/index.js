import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };