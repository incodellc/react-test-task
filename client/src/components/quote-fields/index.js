import QuoteFields from './QuoteFields';
export default QuoteFields;