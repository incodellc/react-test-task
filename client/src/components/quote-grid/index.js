import QuotesGrid from './QuotesGrid';
export default QuotesGrid;