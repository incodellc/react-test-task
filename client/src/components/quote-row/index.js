import QuoteRow from './QuoteRow';
export default QuoteRow;