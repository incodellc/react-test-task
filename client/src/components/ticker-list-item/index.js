import TickerListItem from './TickerListItem';
export default TickerListItem;