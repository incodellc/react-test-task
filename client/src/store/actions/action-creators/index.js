import actionTypes from '../action-types';
import { tickerService } from '../../../services';
import { tickers } from '../../../utils';

export const streamServerConnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_CONNECT
    };
};

export const streamServerDisconnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_DISCONNECT
    };
};

export const updateQuotes = (data) => {
    return {
        type: actionTypes.UPDATE_QUOTES,
        data
    };
};

export const connectToStreamServer = () => {
    return dispatch => {
        dispatch({
            type: actionTypes.CONNECTION_INIT
        });
        tickerService(
            tickers, 
            () => dispatch(streamServerConnect()), 
            (data) => dispatch(updateQuotes(data)),
            () => dispatch(streamServerDisconnect())
        );
    };
};