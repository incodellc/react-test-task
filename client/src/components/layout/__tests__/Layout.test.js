import React from 'react';
import { renderWithRedux } from '../../../utils/test-utils';
import { streamServerConnect, streamServerDisconnect } from '../../../store/actions/action-creators';
import { Layout } from '../../../components';
import { cleanup, fireEvent } from '@testing-library/react';

describe('Check HistoryMenu component', () => {
    afterEach(cleanup);

    it('component renders correctly', () => {
        const { asFragment } = renderWithRedux(<Layout />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('component should contain expected elements and text content', () => {
        const { getByTestId, getByText, store } = renderWithRedux(<Layout><div>Content</div></Layout>);
        expect(getByTestId('layout-menu')).toBeInTheDocument();
        expect(getByTestId('layout-menu')).toContainElement(getByTestId('layout-header'));
        expect(getByTestId('layout-header')).toContainElement(getByTestId('layout-header-icon'));
        expect(getByTestId('layout-header')).toHaveTextContent('React Price Picker');
        expect(getByTestId('layout-header-icon')).toHaveClass('factory');
        expect(getByTestId('layout-menu')).toContainElement(getByTestId('layout-info-button'));
        expect(getByTestId('layout-body')).toBeInTheDocument();
        expect(getByTestId('layout-body')).toContainElement(getByTestId('info-bar'));
        expect(getByTestId('layout-body')).toContainElement(getByTestId('layout-body-dimmer'));
        store.dispatch(streamServerConnect());
        expect(getByTestId('layout-body')).toContainElement(getByText('Content'));
        store.dispatch(streamServerDisconnect());
        expect(getByTestId('layout-body')).toContainElement(getByText('Content'));
    });

    it('check info button content and behavior', () => {
        const { getByTestId } = renderWithRedux(<Layout />);
        expect(getByTestId('layout-info-button')).toHaveTextContent('Info');
        expect(getByTestId('layout-info-button')).toContainElement(getByTestId('layout-info-button-info-icon'));
        expect(getByTestId('info-bar')).not.toHaveClass('visible');
        fireEvent.click(getByTestId('layout-info-button'));
        expect(getByTestId('layout-info-button')).toHaveTextContent('Info');
        expect(getByTestId('layout-info-button')).toContainElement(getByTestId('layout-info-button-close-icon'));
        expect(getByTestId('info-bar')).toHaveClass('visible');
        fireEvent.click(getByTestId('layout-info-button'));
        expect(getByTestId('layout-info-button')).toHaveTextContent('Info');
        expect(getByTestId('layout-info-button')).toContainElement(getByTestId('layout-info-button-info-icon'));
        expect(getByTestId('info-bar')).not.toHaveClass('visible');
    });
});