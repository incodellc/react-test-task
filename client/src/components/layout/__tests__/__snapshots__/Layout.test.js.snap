// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check HistoryMenu component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui borderless top fixed menu menu"
    data-testid="layout-menu"
  >
    <div
      class="item"
    >
      <h3
        class="ui disabled header"
        data-testid="layout-header"
      >
        <i
          aria-hidden="true"
          class="factory icon"
          data-testid="layout-header-icon"
        />
        React Price Picker
      </h3>
    </div>
    <div
      class="right menu"
    >
      <div
        class="item"
      >
        <a
          class="ui basic circular button"
          data-testid="layout-info-button"
          role="button"
        >
          <i
            aria-hidden="true"
            class="info icon"
            data-testid="layout-info-button-info-icon"
          />
          Info
        </a>
      </div>
    </div>
  </div>
  <div
    class="pushable pushable"
    data-testid="layout-body"
  >
    <div
      class="ui segment ui overlay top sidebar infoBar"
      data-testid="info-bar"
    >
      <div
        class="ui container"
      >
        <div
          class="ui stackable grid"
        >
          <div
            class="three wide column"
          />
          <div
            class="five wide column"
          >
            <h4
              class="ui header"
            >
              Contacts
            </h4>
            <div
              class="ui list"
              role="list"
            >
              <div
                class="item"
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  class="user icon"
                />
                <div
                  class="content"
                >
                  Kirill Kopitsa
                </div>
              </div>
              <div
                class="item"
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  class="marker icon"
                />
                <div
                  class="content"
                >
                  Dnipro, Ukraine
                </div>
              </div>
              <div
                class="item"
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  class="mail icon"
                />
                <div
                  class="content"
                >
                  <a
                    href="mailto:joseff.knecht@gmail.com"
                  >
                    joseff.knecht@gmail.com
                  </a>
                </div>
              </div>
              <div
                class="item"
                role="listitem"
              >
                <i
                  aria-hidden="true"
                  class="github icon"
                />
                <div
                  class="content"
                >
                  <a
                    href="https://github.com/joseffone"
                  >
                    github.com/joseffone
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="five wide column"
          >
            <h4
              class="ui header"
            >
              About project
            </h4>
            <p>
              This site is a solution of the test task from InCode Group.
            </p>
          </div>
          <div
            class="three wide column"
          />
        </div>
      </div>
    </div>
    <div
      class="pusher"
    >
      <div
        class="ui active transition visible inverted dimmer"
        data-testid="layout-body-dimmer"
        style="display: flex;"
      >
        <div
          class="content"
        >
          <div
            class="ui inverted text loader"
          >
            Connecting...
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
